django_hosts/resolvers.py

```diff
@@ -172,16 +172,31 @@ def reverse(viewname, args=None, kwargs=None, prefix=None, current_app=None,
     :raises django.core.urlresolvers.NoReverseMatch: if no host or path matches
     :rtype: the fully qualified URL with path
     """
-    host = get_host(host)
-    hostname = reverse_host(host,
-                            args=host_args,
-                            kwargs=host_kwargs)
-    path = reverse_path(viewname,
-                        urlconf=host.urlconf,
-                        args=args or (),
-                        kwargs=kwargs or {},
-                        prefix=prefix,
-                        current_app=current_app)
+    # Try all hosts for 3rd party compatibility.
+    if not (getattr(settings, 'HOST_STICKY', True) or (host or host_args or host_kwargs)):
```

Comment:
also can you turn this around, the code first checks with not, while it could be turned around easily